tests/installation/partitioning/msdos_partition_table.pm

```diff
@@ -22,11 +22,14 @@ sub run {
     my $partitioner = $testapi::distri->get_expert_partitioner;
     $partitioner->run_expert_partitioner;
     foreach my $disk (@{$test_data->{disks}}) {
-        $partitioner->create_new_partition_table({disk => $disk->{name},
-                table_type => $disk->{table_type}, accept_deleting_current_devices_warning => 1});
+        $partitioner->create_new_partition_table({name => $disk->{name},
+                table_type => $disk->{table_type}, accept_deleting_current_devices_warning => 0});
         foreach my $partition (@{$disk->{partitions}}) {
             $partitioner->add_partition_msdos({disk => $disk->{name},
-                    partition => $partition});
+                    partition      => $partition,
```

Comment:
Ok, looks like this change is not needed either, as `$partition` is a hash and contains all the keys you have added explicitly. Please revert it and try again.

Reply:
tried reverting it, then nothing works.

Reply:
Ok, commented to fast, you should just merge keys here, so  it should be:
```
 $partitioner->add_partition_msdos({disk => $disk->{name}, %{$partition} });
```

Reply:
I just needed to add $args->{partition_type} in add_partition_msdos and remove role and size from here, now all works :)